main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapActivity.java

```diff
@@ -418,22 +416,34 @@ private void reloadCachesAndWaypoints() {
                                 onReceiveTargetUpdate(new AbstractDialogFragment.TargetInfo(waypoint.getCoords(), waypoint.getName()));
                             }
                         }
-                    } else if (cache.getCoords() != null) { // geocache mode: display ONLY geocache and its waypoints
-                        viewModel.caches.write(false, c -> {
-                            c.clear();
-                            c.add(cache);
-                        });
-                        if (setDefaultCenterAndZoom) {
-                            mapFragment.setCenter(cache.getCoords());
-                        }
-                        viewModel.waypoints.write(wps -> {
-                            wps.clear();
-                            final Set<Waypoint> waypoints = new HashSet<>(cache.getWaypoints());
-                            MapUtils.filter(waypoints, getFilterContext());
-                            wps.addAll(waypoints);
-                        });
-                        if (!isTargetSet()) {
-                            onReceiveTargetUpdate(new AbstractDialogFragment.TargetInfo(cache.getCoords(), cache.getGeocode()));
+                    } else { // geocache mode: display ONLY geocache and its waypoints, if coordinates are available
+                        final Geopoint cacheCoordinates = cache.getCoords();
+                        final Waypoint firstWaypoint = cacheCoordinates == null ? cache.getFirstMatchingWaypoint(wp -> wp.getCoords() != null) : null;
+                        final Geopoint currentCoordinates = null != firstWaypoint ? firstWaypoint.getCoords() : cacheCoordinates;
+
+                        if (null != currentCoordinates) {
+                            viewModel.caches.write(false, c -> {
+                                c.clear();
+                                c.add(cache);
```

Comment:
This means there could be a cache here which has no coordinates. Doesn't this cause any problemss later?

Reply:
Good point...
checked usages:
* getClosestDistanceInM is used on the caches, but it checks for null-coordinates
* is used to create markers and distance-circles with GeoPrimitive.createMarker and GeoPrimitive.createCircles-> no null-check, null added to a list, also the other GeoPrimitives uses a Geopoint without null-check.

Will check those scenarios in emulator with the current implementation.
Will check same scenarios in emulator with not added a cache without coordinates.